--- pristine/http-parser-0.9.0/http_parser/parser.c	2020-04-07 06:24:00
+++ http-parser-0.9.0/http_parser/parser.c	2026-08-17 21:45:44
@@ -147,7 +147,7 @@
     #undef CYTHON_USE_PYLONG_INTERNALS
     #define CYTHON_USE_PYLONG_INTERNALS 0
   #elif !defined(CYTHON_USE_PYLONG_INTERNALS)
-    #define CYTHON_USE_PYLONG_INTERNALS 1
+    #define CYTHON_USE_PYLONG_INTERNALS 0
   #endif
   #ifndef CYTHON_USE_PYLIST_INTERNALS
     #define CYTHON_USE_PYLIST_INTERNALS 1
@@ -174,7 +174,11 @@
     #define CYTHON_FAST_THREAD_STATE 1
   #endif
   #ifndef CYTHON_FAST_PYCALL
-    #define CYTHON_FAST_PYCALL 1
+    #if PY_VERSION_HEX >= 0x030B0000
+      #define CYTHON_FAST_PYCALL 0
+    #else
+      #define CYTHON_FAST_PYCALL 1
+    #endif
   #endif
   #ifndef CYTHON_PEP489_MULTI_PHASE_INIT
     #define CYTHON_PEP489_MULTI_PHASE_INIT (PY_VERSION_HEX >= 0x03050000)
@@ -314,8 +318,13 @@
   #define __Pyx_PyCode_New(a, k, l, s, f, code, c, n, v, fv, cell, fn, name, fline, lnos)\
           PyCode_New(a, 0, k, l, s, f, code, c, n, v, fv, cell, fn, name, fline, lnos)
 #else
+  #if PY_VERSION_HEX >= 0x030B0000
   #define __Pyx_PyCode_New(a, k, l, s, f, code, c, n, v, fv, cell, fn, name, fline, lnos)\
+          PyCode_NewWithPosOnlyArgs(a, 0, k, l, s, f, code, c, n, v, fv, cell, fn, name, name, fline, lnos, __pyx_empty_bytes)
+  #else
+  #define __Pyx_PyCode_New(a, k, l, s, f, code, c, n, v, fv, cell, fn, name, fline, lnos)\
           PyCode_New(a, k, l, s, f, code, c, n, v, fv, cell, fn, name, fline, lnos)
+  #endif
 #endif
   #define __Pyx_DefaultClassType PyType_Type
 #endif
@@ -366,7 +375,11 @@
   #define __Pyx_PyFrame_SetLineNumber(frame, lineno) PyFrame_SetLineNumber(frame, lineno)
 #else
   #define __Pyx_PyCode_HasFreeVars(co)  (PyCode_GetNumFree(co) > 0)
+  #if PY_VERSION_HEX >= 0x030B0000
+  #define __Pyx_PyFrame_SetLineNumber(frame, lineno)  (void)0
+  #else
   #define __Pyx_PyFrame_SetLineNumber(frame, lineno)  (frame)->f_lineno = (lineno)
+  #endif
 #endif
 #if !CYTHON_FAST_THREAD_STATE || PY_VERSION_HEX < 0x02070000
   #define __Pyx_PyThreadState_Current PyThreadState_GET()
@@ -1168,7 +1181,7 @@
     if (likely(L->allocated > len) & likely(len > (L->allocated >> 1))) {
         Py_INCREF(x);
         PyList_SET_ITEM(list, len, x);
-        Py_SIZE(list) = len+1;
+        Py_SET_SIZE(list, len+1);
         return 0;
     }
     return PyList_Append(list, x);
@@ -8470,8 +8483,13 @@
 __Pyx_PyErr_GetTopmostException(PyThreadState *tstate)
 {
     _PyErr_StackItem *exc_info = tstate->exc_info;
+#if PY_VERSION_HEX >= 0x030B0000
+    while ((exc_info->exc_value == NULL || exc_info->exc_value == Py_None) &&
+           exc_info->previous_item != NULL)
+#else
     while ((exc_info->exc_type == NULL || exc_info->exc_type == Py_None) &&
            exc_info->previous_item != NULL)
+#endif
     {
         exc_info = exc_info->previous_item;
     }
@@ -8484,9 +8502,22 @@
 static CYTHON_INLINE void __Pyx__ExceptionSave(PyThreadState *tstate, PyObject **type, PyObject **value, PyObject **tb) {
     #if CYTHON_USE_EXC_INFO_STACK
     _PyErr_StackItem *exc_info = __Pyx_PyErr_GetTopmostException(tstate);
+    #if PY_VERSION_HEX >= 0x030B0000
+    *value = exc_info->exc_value;
+    Py_XINCREF(*value);
+    if (*value == NULL) {
+        *type = NULL;
+        *tb = NULL;
+    } else {
+        *type = (PyObject *) PyExceptionInstance_Class(*value);
+        Py_XINCREF(*type);
+        *tb = PyException_GetTraceback(*value);
+    }
+    #else
     *type = exc_info->exc_type;
     *value = exc_info->exc_value;
     *tb = exc_info->exc_traceback;
+    #endif
     #else
     *type = tstate->exc_type;
     *value = tstate->exc_value;
@@ -8497,6 +8528,15 @@
     Py_XINCREF(*tb);
 }
 static CYTHON_INLINE void __Pyx__ExceptionReset(PyThreadState *tstate, PyObject *type, PyObject *value, PyObject *tb) {
+    #if PY_VERSION_HEX >= 0x030B0000
+    PyObject *tmp_value;
+    _PyErr_StackItem *exc_info = tstate->exc_info;
+    tmp_value = exc_info->exc_value;
+    exc_info->exc_value = value;
+    Py_XDECREF(tmp_value);
+    Py_XDECREF(type);
+    Py_XDECREF(tb);
+    #else
     PyObject *tmp_type, *tmp_value, *tmp_tb;
     #if CYTHON_USE_EXC_INFO_STACK
     _PyErr_StackItem *exc_info = tstate->exc_info;
@@ -8517,6 +8557,7 @@
     Py_XDECREF(tmp_type);
     Py_XDECREF(tmp_value);
     Py_XDECREF(tmp_tb);
+    #endif
 }
 #endif
 
@@ -8629,7 +8670,7 @@
 {
     PyObject *local_type, *local_value, *local_tb;
 #if CYTHON_FAST_THREAD_STATE
-    PyObject *tmp_type, *tmp_value, *tmp_tb;
+    PyObject *tmp_type = NULL, *tmp_value = NULL, *tmp_tb = NULL;
     local_type = tstate->curexc_type;
     local_value = tstate->curexc_value;
     local_tb = tstate->curexc_traceback;
@@ -8662,12 +8703,17 @@
     #if CYTHON_USE_EXC_INFO_STACK
     {
         _PyErr_StackItem *exc_info = tstate->exc_info;
+        #if PY_VERSION_HEX >= 0x030B0000
+        tmp_value = exc_info->exc_value;
+        exc_info->exc_value = local_value;
+        #else
         tmp_type = exc_info->exc_type;
         tmp_value = exc_info->exc_value;
         tmp_tb = exc_info->exc_traceback;
         exc_info->exc_type = local_type;
         exc_info->exc_value = local_value;
         exc_info->exc_traceback = local_tb;
+        #endif
     }
     #else
     tmp_type = tstate->exc_type;
